spec/services/journey_logger_service_spec.rb

```diff
@@ -206,5 +206,67 @@
         expect(output.early_eligibility_result).to be true
       end
     end
+
+    context "when it is an early ineligible result for gross income", :ee_banner do
+      let(:session_data) do
+        {
+          immigration_or_asylum: false,
+          partner: false,
+          passporting: false,
+          early_result: { "result" => "ineligible",
+                          "type" => "gross_income" },
+        }.with_indifferent_access
+      end
+
+      it "saves `early_result_type` and `outcome` to completed user journey" do
+        described_class.call(assessment_id, calculation_result, check, portal_user_office_code, {})
+        output = CompletedUserJourney.find_by(assessment_id:)
+        expect(output.asylum_support).to be false
+        expect(output.partner).to be false
+        expect(output.passported).to be false
+        expect(output.outcome).to eq "ineligible"
+        expect(output.early_result_type).to eq "gross_income"
+        expect(output.early_eligibility_result).to be true
+      end
+    end
+
+    context "when details change in a full journey", :ee_banner do
```

Comment:
I dont think these 3 tests are actually testing the change that we have made, although it runs over the code, it is creating a new record but without focussing on changing the `early_eligibility_result`. I have added a test that mimics what we do in the service and shows how we are creating 2 records for the same `assessment_id`. Which I think can replace these 3 contexts.

Reply:
My only question on the added test:

- do we not need to add some session data, before we make the initial call, somewhere before line 234?
- do we want to check the output of what we think should be in `CompletedUserJourney`